src/qz/printer/action/WebApp.java

```diff
@@ -131,28 +141,30 @@ public WebApp() {
     /** Starts JavaFX thread if not already running */
     public static synchronized void initialize() throws IOException {
         if (instance == null) {
+            startupLatch = new CountDownLatch(1);
+
             new Thread() {
                 public void run() {
                     Application.launch(WebApp.class);
                 }
             }.start();
         }
 
-        for(int i = 0; i < (TIMEOUT * 1000); i += SLEEP) {
-            if (started.get()) { break; }
-
-            log.trace("Waiting for JavaFX..");
-            try { Thread.sleep(SLEEP); } catch(Exception ignore) {}
-        }
+        if (startupLatch.getCount() > 0) {
+            try {
+                log.trace("Waiting for JavaFX..");
 
-        if (!started.get()) {
-            throw new IOException("JavaFX did not start");
+                if (!startupLatch.await(1000, TimeUnit.MILLISECONDS)) {
```

Comment:
I think this should be `60, TimeUnit.SECONDS` to match our existing code of `TIMEOUT = 60; //total paused seconds before failing`

Reply:
@bberenz bump as I don't think this snippet was ever resolved.

Reply:
I thought I addressed it at some point, but it must have gotten reverted in one of the rollbacks. Taken care of now.